components/suggestion/suggestion_box.jsx

```diff
@@ -374,6 +380,22 @@ export default class SuggestionBox extends React.Component {
         this.refs.input.blur();
     }
 
+    setContainerRef = (container) => {
+        // Attach/detach event listeners that aren't supported by React
+        if (this.container) {
+            this.container.removeEventListener('focusin', this.handleFocusIn);
+            this.container.removeEventListener('focusout', this.handleFocusOut);
+        }
+
+        if (container) {
+            container.addEventListener('focusin', this.handleFocusIn);
+            container.addEventListener('focusout', this.handleFocusOut);
```

Comment:
I believe this should be enough since nothing outside of the SuggestionBox is holding a reference to the component or the listener, so they'll get cleaned up. According to [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Memory_Management#Mark-and-sweep_algorithm), all modern browsers use a garbage collector where the issue of the SuggestionBox holding a reference to the element and vice versa shouldn't be an issue